Hof 405

DivineTaminangHO opened the pull request:
- Anchore replaced with trivy scan in the Drone.yaml
- The node image updated from node:14 to node:18
- The cve-exception file updated

DivineTaminangHO committed:
Update .drone.yml

What?: I have replaced anchore  with Trivy scan
why?: This is because the ACP team has now decided to use Trivy instead of Anchore scans. Anchore has not been supported since January 2023 and as result runs the risk of not detecting vulnerabilities effectively.  
Anchore is an open source scanner which scans the following: OS packages and software dependencies in use (SBOM)
Known vulnerabilities (CVEs)
IaC issues and misconfigurations
Sensitive information and secrets
Software licenses
https://docs.acp.homeoffice.gov.uk/how-to/security/trivy/#trivy-overview
How?:
Trivy has been added to drone.yaml pipeline which will scan against any vulnerability to primarily detect CVEs present in OS and Language packages. In addition it will try to detect any secrets like private rsa keys within the container image layers.

DivineTaminangHO committed:
Update and rename cve-exceptions.txt to trivy-cve-exceptions.txt

adityababumallisettiHO commented on .drone.yml:
```diff
@@ -24,14 +24,30 @@ steps:
       - n=0; while [ "$n" -lt 60 ] && [ ! docker stats --no-stream ]; do n=$(( n + 1 )); sleep 1; done
       - docker build -t html-pdf-converter .
 
-  - name: scan-image
```
Trivy Scan step should run after Docker push to Quay.
Also Image name has to be set to quay.io/ukhomeofficedigital/html-pdf-converter:$${DRONE_TAG} 
Please follow ETA .drone.yml
We need these added 
      SEVERITY: MEDIUM,HIGH,CRITICAL
      FAIL_ON_DETECTION: false
      IGNORE_UNFIXED: true
and
ALLOW_CVE_LIST_FILE: trivy-cve-exceptions.txt
Also please check the conditions of Drone that executes the drone pipeline.

adityababumallisettiHO reviewed (commented):
Scan-image step has to run after the Image has been pushed to the Quay.io repo

DivineTaminangHO committed:
Update .drone.yml

Anchore replaced by Trivy scan.
some additional details added to trivy scan steps

DivineTaminangHO committed:
Update .drone.yml

Corrected some environmental variable for the image-to-quay step

DivineTaminangHO commented:
What?: I have replaced anchore  with Trivy scan
why?: This is because the ACP team has now decided to use Trivy instead of Anchore scans. Anchore has not been supported since January 2023 and as result runs the risk of not detecting vulnerabilities effectively.  
Anchore is an open source scanner which scans the following: OS packages and software dependencies in use (SBOM)
Known vulnerabilities (CVEs)
IaC issues and misconfigurations
Sensitive information and secrets
Software licenses
https://docs.acp.homeoffice.gov.uk/how-to/security/trivy/#trivy-overview
How?:
Trivy has been added to drone.yaml pipeline which will scan against any vulnerability to primarily detect CVEs present in OS and Language packages. In addition it will try to detect any secrets like private rsa keys within the container image layers.

DivineTaminangHO committed:
Update .drone.yml

DivineTaminangHO committed:
Update .drone.yml

Code edited with commented sections for anchore removed

adityababumallisettiHO reviewed (commented):
Please correct the indentations.

DivineTaminangHO committed:
Update .drone.yml

indentitions changes

DivineTaminangHO committed:
Update .drone.yml

changes to scan-image yaml files' intentition

adityababumallisettiHO commented:
Please update Git commits to HOF standards.

This link helps you : https://cbea.ms/git-commit/#separate

DivineTaminangHO committed:
Update .drone.yml

indentation corrected
